errai-config/src/main/java/org/jboss/errai/config/propertiesfile/ErraiAppPropertiesErraiModulesConfiguration.java

```diff
@@ -0,0 +1,349 @@
+
+        final ResourceBundle props = new PropertyResourceBundle(inputStream);
+
+        for (final String key : props.keySet()) {
+          final String value = props.getString(key);
+
+          if (key.equals(ErraiAppPropertiesErraiModulesConfiguration.SERIALIZABLE_TYPES)) {
+            addSerializableTypes(exposedClasses, value);
+          } else if (key.equals(ErraiAppPropertiesErraiModulesConfiguration.NON_SERIALIZABLE_TYPES)) {
+            addNonSerializableTypes(exposedClasses, nonportableClasses, value);
+          }
+        }
+      } catch (final IOException e) {
+        throw new RuntimeException("error reading ErraiApp.properties", e);
+      } finally {
+        if (inputStream != null) {
+          try {
+            inputStream.close();
+          } catch (final IOException e) {
+            //
+          }
+        }
+      }
+    }
+  }
+
+  private static void addNonSerializableTypes(final Set<MetaClass> exposedClasses,
+          final Set<MetaClass> nonportableClasses,
+          final String value) {
+
+    final Set<String> patterns = new LinkedHashSet<>();
+    for (final String s : value.split(" ")) {
+      final String singleValue = s.trim();
+      if (singleValue.endsWith("*")) {
+        patterns.add(singleValue);
+      } else {
+        try {
+          nonportableClasses.add(MetaClassFactory.get(singleValue));
+        } catch (final Exception e) {
+          throw new RuntimeException("could not find class defined in ErraiApp.properties as nonserializable: " + s, e);
+        }
+      }
+    }
+    if (!patterns.isEmpty()) {
+      final SimplePackageFilter filter = new SimplePackageFilter(patterns);
+      MetaClassFactory.getAllCachedClasses()
+              .stream()
+              .filter(mc -> filter.apply(mc.getFullyQualifiedName()))
+              .collect(toCollection(() -> exposedClasses));
+    }
+  }
+
+  private static void addSerializableTypes(final Set<MetaClass> exposedClasses, final String value) {
+    final Set<String> patterns = new LinkedHashSet<>();
+    for (final String s : value.split(" ")) {
+      final String singleValue = s.trim();
+      if (singleValue.isEmpty()) {
+        continue;
+      }
+      if (singleValue.endsWith("*")) {
+        patterns.add(singleValue);
+      } else {
+        try {
+          exposedClasses.add(MetaClassFactory.get(singleValue));
+        } catch (final Exception e) {
+          throw new RuntimeException("could not find class defined in ErraiApp.properties for serialization: " + s, e);
+        }
+      }
+    }
+    if (!patterns.isEmpty()) {
+      final SimplePackageFilter filter = new SimplePackageFilter(patterns);
+      MetaClassFactory.getAllCachedClasses()
+              .stream()
+              .filter(mc -> filter.apply(mc.getFullyQualifiedName()) && validateWildcardSerializable(mc))
+              .collect(toCollection(() -> exposedClasses));
+    }
+  }
+
+  private static boolean validateWildcardSerializable(MetaClass mc) {
```

Comment:
Yeah, I saw that PR too. Actually, I think we should rebase errai/master in errai/errai-codegen-apt. I'll solve the conflicts this PR will have so that this PR will contain the new configuration property in both ErraiApp.properties and @ErraiApp. What do you think?

Reply:
Not sure I completely understand. Are you suggesting we deal with this when merging this branch into master?

Reply:
No, I think we should run these commands before merging this PR.

`git checkout errai-codegen-apt`
`git fetch origin`
`git rebase master`
`git push -f origin errai-codegen-apt`

Then almost certainly this branch (`tiagobento/ERRAI-1075.4`) will have conflicts. I solve them and we merge this PR. What do you think?

Reply:
I think you will have to do a git merge instead, otherwise you need to rewrite history which I don't think works with pull requests. Otherwise I agree. Just be careful, because it might be possible that merging works without conflicts if you deleted the file that the changes on master are in.

Reply:
Alright, I'll do a merge then :)